File: ruby/Proxy/movie_player.rb
#this class acts as Proxy
require_relative 'video_functions'

class MoviePlayer < _________________ # <<1
  attr_accessor :movie
  attr_accessor :name

  def initialize(name)
    @movie = Movie.new(name)
  end

  def playMovie
    _________________ # <<2
  end

  def getMovieDetails
    movie.getMovieDetails
  end

  def deleteMovie
    puts "\nPermission denied!"
  end
end

File: java/Proxy/src/MoviePlayer.java
package exercise;


// this class acts as Proxy
class MoviePlayer ________________ //>> 1 {

	Movie movie;
	String name;

	public MoviePlayer(String name) {
		super();
		movie = new Movie(name);
	}

	public void playMovie() {
		________________ //>> 2
	}

	public String getMovieDetails() {
		return movie.getMovieDetails();
	}
	
	public void deleteMovie(String movieName) {
		System.out.println("Permission denied!!");
	}
}